### DISCUSSION TITLE
Add `docker-compose.yml` to demo payload-3.0-demo + container image releases

### DISCUSSION THREAD

**Discussion by jo-chemla** [payloadcms/payload#8114] [Feature Requests & Ideas]
Hi there, 
Just discovered the [v3-preview](https://payloadcms.com/v3-preview) (+ [repo](https://github.com/payloadcms/payload-3.0-demo)) and the fact that your team want to make payload-v3 a full-featured CMS + frontend as stated on this nextjs [reddit-thread](https://www.reddit.com/r/nextjs/comments/1cxkbil/fullstack_web_application_similar_to_wordpress/). Pretty exciting stuff! 

Would love to make it easier to self-host, e.g. by providing an example `docker-compose.yml` in addition to [instructions](https://github.com/payloadcms/payload-3.0-demo#using-this-repo) and pre-built container image on ghrc.io or docker-hub, which would be akin to running the steps listed on your readme - roughly `.env.local` + `pnpm i` + `.\start-database.sh` + `pnpm dev`. 

Might be too soon given this is still early development, but would make it easier to try-out-v3 and receive early-adopters feedback! Thanks!

**Comment by jimmyharika** [comment]
> Hi there, Just discovered the [v3-preview](https://payloadcms.com/v3-preview) (+ [repo](https://github.com/payloadcms/payload-3.0-demo)) and the fact that your team want to make payload-v3 a full-featured CMS + frontend as stated on this nextjs [reddit-thread](https://www.reddit.com/r/nextjs/comments/1cxkbil/fullstack_web_application_similar_to_wordpress/). Pretty exciting stuff!
> 
> Would love to make it easier to self-host, e.g. by providing an example `docker-compose.yml` in addition to [instructions](https://github.com/payloadcms/payload-3.0-demo#using-this-repo) and pre-built container image on ghrc.io or docker-hub, which would be akin to running the steps listed on your readme - roughly `.env.local` + `pnpm i` + `.\start-database.sh` + `pnpm dev`.
> 
> Might be too soon given this is still early development, but would make it easier to try-out-v3 and receive early-adopters feedback! Thanks!

Ditto Problem

**Reply by JP-HoneyBadger** [reply]
I am trying to figure out docker-compose currently myself.  I just built a new prod server and want to use docker containers instead of VMware as the current prod server runs.

**Comment by markusahrweileramcon** [comment]
So, I have used next without payload 3 before and had no workaround, to build the app other than in the deployed container itself.
Payload needs a connection to a database to boot and with payload 3 (default setup) the build process needs a running payload to fetch the page data.

So, the documentation on https://payloadcms.com/docs/production/deployment#docker seems to be wrong, since in a production environment you would probably not build the container on the same server, which contains the database. Or the database has to be available to the build server which also should not be the case in production.

I changed the port, nextjs is using to start the server while building to be 3001 and set the readiness probe on the container in k8s to listen for port 3000 after 5 minutes.
This takes longer to deploy (redeploy) but does give a result which has static site rendering.

The Payload 3 app does not use static sites, but still needs a database connection on build, which seems a bit weird to me.

So I would like to hear from a payload team member, if there is any other solution, if you are not building your application on the production server

**Reply by TheRobBrennan** [reply]
I'm in the same boat. Without having a database to build against, it does crash and is a problem. It also makes me wonder the best patterns/practices for handling new code that involves schema changes and migrations - and how that would look migrating from a dev/staging environment to production.
